src/backend/distributed/commands/dependencies.c

```diff
@@ -95,7 +95,7 @@ EnsureDependenciesExistOnAllNodes(const ObjectAddress *target)
 	 * either get it now, or get it in citus_add_node after this transaction finishes and
 	 * the pg_dist_object record becomes visible.
 	 */
-	List *workerNodeList = ActivePrimaryNonCoordinatorNodeList(RowShareLock);
+	List *workerNodeList = ActivePrimaryRemoteNodeList(RowShareLock);
```

Comment:
I have not read the code, but just wanted to let you know that @hanefi has a good context on the intricacies of DDL from any node: #7009, probably worth checking with him.

Reply:
Yeah I'll talk to @hanefi when is back, thank you @onderkalaci :)